.github/workflows/build.yml

```diff
@@ -0,0 +1,28 @@
+name: Build Frontend
+
+on:
+  pull_request:
+    branches: [ main ]
+  push:
+    branches: [ main ]
+
+jobs:
+  checkout-lint-build:
+  
+    runs-on: ubuntu-latest
+    concurrency: ci-${{ github.ref }}
+     
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v2
+      
+      - name: Lint (TS)
+        uses: icrawl:action-eslint@v1
+      
+      - name: Build
+        run: |
+              yarn install
```

Comment:
Generally in CICD it's best practice to have reproducible builds - `yarn install` by default checks for updates, and may install different versions than the ones specified in the lockfile (e.g. causing build to fail unexpectedly). To prevent this, you can use the `--frozen-lockfile` argument.